server/api.js

```diff
@@ -343,13 +343,16 @@ router.get("/avr/:channelId/:userId", async (req, res) => {
 
 router.get("/channelAvg/:channelId", (req, res) => {
 	const channelId = req.params.channelId;
+	console.log(channelId)
 
 	const query = `SELECT channel_id, AVG(message_count)::numeric(10,1) AS avg_message, AVG(reaction_count)::numeric(10,1) AS avg_reaction FROM messages WHERE date > current_date - interval '7 days' AND channel_id = '${channelId}' GROUP BY channel_id ORDER by channel_id`;
```

Comment:
I think by using AVG function , it calculates average based number of rows in table not weekly average, so it divides by number of user* number of days instead of only number of days

Reply:
Thanks Diba for the review. There is WHERE condition in the query to get only last 7 seven days to calculate AVG of the last week.

Reply:
Yes, but there are numbers of row for everyday depends on number of user_id

Reply:
This query isn't quite correct - it will be giving the average per user per day (since there is a row per user per day). So it needs to be changed a bit. I can help later today.

If we want the average for each week rather than the previous week see https://stackoverflow.com/questions/10424395/how-do-i-group-a-date-field-by-week-where-the-weeks-start-on-saturday